draft-ietf-jsonpath-base.md

```diff
@@ -1270,14 +1273,15 @@ rather than the negated existence test `!@.foo` (which yields false if `@.foo` s
 
 The comparison operators `==` and `<` are defined first and then these are used to define `!=`, `<=`, `>`, and `>=`.
 
-When a path resulting in an empty nodelist appears on either side of a comparison:
+When a path or function expression resulting in an empty nodelist appears on either side of a comparison:
 
 * a comparison using the operator `==` yields true if and only if the comparison
-is between two paths each of which result in an empty nodelist.
+is between two `comparable`s, each of which is a path or a function expression, and each of which result in an empty nodelist.
```

Comment:
I think the context of line 1276 allows us to simplify this

> When a path or function expression resulting in an empty nodelist appears on either side of a comparison:
> 
> * a comparison using the operator `==` yields true if and only if both operands, whether path or function expression, result in an empty nodelist.

We already have from the 1276 the condition that _one_ side is an empty nodelist.  We just need to say that the other one is as well.

Nit: I don't really care for "function _expression_."  I much prefer just "function."  The whole thing is an expression that can contain a function.

Reply:
> I think the context of line 1276 allows us to simplify this
> 
> > When a path or function expression resulting in an empty nodelist appears on either side of a comparison:
> > 
> > * a comparison using the operator `==` yields true if and only if both operands, whether path or function expression, result in an empty nodelist.
> 
> We already have from the 1276 the condition that _one_ side is an empty nodelist. We just need to say that the other one is as well.

Nice. Adopted.

> 
> Nit: I don't really care for "function _expression_." I much prefer just "function." The whole thing is an expression that can contain a function.

We have to be careful to distinguish between a function and where it is used. Also I wouldn't want `length(@.a) == 17`, for example, to be regarded as a function expression. (I considered "function call" instead of "function expression", but that has  operational connotations which aren't helpful.)

Reply:
> > Essentially, it says that comparisons of the form `<empty nodelist> == ...` or `... == <empty nodelist>` yield true if and only if both sides are `<empty nodelist>`.
> 
> You never get to compare nodelists. You always compare comparables, which are ValueType. These have "Nothing", not "empty nodelist".

Yes, I was being a bit sloppy there. But I'd stop short of saying that comparables are ValueType at this point. See my other comment about rolling out the type system across the spec.

Reply:
> On 1. Mar 2023, at 09:35, Glyn Normington ***@***.***> wrote: We don't need to define exist() and value() now because the pressing use cases for these functions have been dispensed with by this PR.
> The problem is that not having these defined upfront makes any design of a function extension likely to contort around not having them. Grüße, Carsten

Ok, I can see the value of adding some such functions, but I'd want to consider them carefully and make them the subject of a separate issue/PR. I don't think we need them in this PR.